From 11c1fe24ade6a2f8776b9e84432dedeba9132adf Mon Sep 17 00:00:00 2001
From: jerome <jerome.anand@intel.com>
Date: Wed, 30 Jan 2013 13:29:33 +0530
Subject: [PORT FROM MAIN][TINY-ALSA] Support for 24 bit LE format

BZ: 81843

The pcm lib is modified to add support for
24 bit little endian.

Orig-Change-Id: I5ae222baaa3a37351b349da8e716937a1edd5ad1
Signed-off-by: jerome <jerome.anand@intel.com>

Change-Id: I2bf38d1d22bdff1977fbcb8d2f7f48490cb195ab
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 include/tinyalsa/asoundlib.h |    2 +-
 pcm.c                        |    1 +
 2 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/include/tinyalsa/asoundlib.h b/include/tinyalsa/asoundlib.h
index 6346da4..6333e7a 100644
--- a/include/tinyalsa/asoundlib.h
+++ b/include/tinyalsa/asoundlib.h
@@ -70,9 +70,9 @@ struct pcm;
 /* Bit formats */
 enum pcm_format {
     PCM_FORMAT_S16_LE = 0,
+    PCM_FORMAT_S24_LE,
     PCM_FORMAT_S32_LE,
     PCM_FORMAT_S8,
-    PCM_FORMAT_S24_LE,
 
     PCM_FORMAT_MAX,
 };
diff --git a/pcm.c b/pcm.c
index d80b20d..5c41f7d 100644
--- a/pcm.c
+++ b/pcm.c
@@ -217,6 +217,7 @@ static unsigned int pcm_format_to_bits(enum pcm_format format)
 {
     switch (format) {
     case PCM_FORMAT_S32_LE:
+    case PCM_FORMAT_S24_LE:
         return 32;
     default:
     case PCM_FORMAT_S16_LE:
-- 
1.7.4.1

